### PR TITLE
Added config for mail encryption

### PR DESCRIPTION
**Pull request by dhee-tree**
### What

- Added `Encrypted` config 
- Only false for containers to prevent unencrypted connection error when receiving post request

### How to review

- CI test passes
- `Encrypted` Is set to `true`
- Sense check 

### Who can review

!Me

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -22,10 +22,11 @@ type Config struct {
...
 type Mail struct {
-	Host     string `envconfig:"MAIL_HOST"`
-	User     string `envconfig:"MAIL_USER"`
-	Password string `envconfig:"MAIL_PASSWORD" json:"-"`
-	Port     string `envconfig:"MAIL_PORT"`
+	Host      string `envconfig:"MAIL_HOST"`
+	User      string `envconfig:"MAIL_USER"`
+	Password  string `envconfig:"MAIL_PASSWORD" json:"-"`
+	Port      string `envconfig:"MAIL_PORT"`
+	Encrypted bool   `envconfig:"ENCRYPT_MAIL"`
```
Comment (lindenmckenzie) [inline config/config.go]:
```suggestion
	Encrypted bool   `envconfig:"MAIL_ENCRYPTION"`
```
for consistency.

File: email/sender.go [hunk 2]
```diff
@@ -27,14 +27,19 @@ func (s SMTPSender) Send(from string, to []string, msg []byte) error {
...
+			cfg.User,
+			cfg.Password,
+			cfg.Host,
+		)
+		if cfg.Host == "localhost" {
+			auth = unencryptedAuth{auth}
+		}
+	} else {
+		auth = smtp.CRAMMD5Auth(cfg.User, cfg.Password)
 	}
```
Comment (lindenmckenzie) [inline email/sender.go]: Is this necessary? wouldn't the following be simpler
```suggestion
	var auth smtp.Auth
	auth = smtp.PlainAuth(
			"",
			cfg.User,
			cfg.Password,
			cfg.Host,
		)
	if !cfg.Encrypted {
	      auth = unencryptedAuth{auth}
	} 
```

**Review by lindenmckenzie — CHANGES_REQUESTED** [review]

File: email/sender.go (hunk 2 above)
Comment (dhee-tree) [reply email/sender.go]: This also throws unencrypted connection, PlainAuth will only send the credentials if the connection is using TLS or is connected to localhost. 

We connect to mailhog in the container [here](https://github.com/ONSdigital/dp-compose/blob/e8d5fcb34e120031c199cf31a8e4bd26ad7b6fad/v2/manifests/core-ons/dp-feedback-api.yml#L25)

File: config/config.go (hunk 1 above)
Comment (dhee-tree) [reply config/config.go]: Done

File: email/sender.go (hunk 2 above)
Comment (lindenmckenzie) [reply email/sender.go]: Sure, makes sense - is there a solution that will work for both localhost and in docker?

Comment (dhee-tree) [reply email/sender.go]: Yes, Encrypted would only be false on docker, then we will use CRAMMD5Auth instead of PlainAuth

Comment (lindenmckenzie) [reply email/sender.go]:
```suggestion
	var auth smtp.Auth
	if cfg.Encrypted {
		auth = smtp.PlainAuth(
			"",
			cfg.User,
			cfg.Password,
			cfg.Host,
		)
	} else {
		auth = smtp.CRAMMD5Auth(cfg.User, cfg.Password)
	}
```

Why not this then?

Comment (dhee-tree) [reply email/sender.go]: Done

**Review by lindenmckenzie — APPROVED** [review]

**Pull request merged**